passkey-client/src/tests/mod.rs

```diff
@@ -101,11 +104,79 @@ async fn create_and_authenticate() {
         public_key: good_credential_request_options(credential_id),
     };
     client
-        .authenticate(&origin, auth_options, None)
+        .authenticate(&origin, auth_options, DefaultClientData)
         .await
         .expect("failed to authenticate with freshly created credential");
 }
 
+#[tokio::test]
+async fn create_and_authenticate_with_extra_client_data() {
+    #[derive(Clone, Serialize, Deserialize)]
+    struct AndroidClientData {
+        android_package_name: String,
+    }
+    let auth = Authenticator::new(
+        ctap2::Aaguid::new_empty(),
+        MemoryStore::new(),
+        uv_mock_with_creation(2),
+    );
+    let mut client = Client::new(auth);
+
+    let origin = Url::parse("https://future.1password.com").unwrap();
+    let options = webauthn::CredentialCreationOptions {
+        public_key: good_credential_creation_options(),
+    };
+    let extra_data = AndroidClientData {
+        android_package_name: "com.example.app".to_owned(),
+    };
+    let cred = client
+        .register(
+            &origin,
+            options,
+            DefaultClientDataWithExtra(extra_data.clone()),
+        )
+        .await
+        .expect("failed to register with options");
+
+    let returned_base64url_client_data_json: String = cred.response.client_data_json.into();
+    let returned_client_data_json =
+        try_from_base64url(returned_base64url_client_data_json.as_str())
+            .expect("could not base64url decode client data");
+    let returned_client_data: CollectedClientData<AndroidClientData> =
+        serde_json::from_slice(&returned_client_data_json)
+            .expect("could not json deserialize client data");
+    assert_eq!(
+        returned_client_data.extra_data.android_package_name,
+        "com.example.app"
+    );
+
+    let credential_id = cred.raw_id;
+
+    let auth_options = webauthn::CredentialRequestOptions {
+        public_key: good_credential_request_options(credential_id),
+    };
+    let result = client
+        .authenticate(
+            &origin,
+            auth_options,
+            DefaultClientDataWithExtra(extra_data.clone()),
```

Comment:
**Nit:** This clone is no longer necessary

Reply:
True, it's cloned further up so the original object is available here. Removed!